src/auth/userSession.ts

```diff
@@ -398,4 +400,58 @@ export class UserSession {
 
     return gaiaConfig
   }
+
+  /**
+   * Retreive a collection Gaia hub config from local user data
+   * @param collection - the name of the collection to get hub config for
+   */
+  getCollectionGaiaHubConnection(collectionName: string): Promise<GaiaHubConfig> {
```

Comment:
I think there is a bug somewhere around here where the non-collection bucket gets used after a Gaia storage request fails, and the error handler attempts to re-negotiate the gaia hub connection. 

Saw this when testing collection list files while using a faulty gaia hub server.